tests/test_ec2/test_instances.py

```diff
@@ -2057,15 +2078,45 @@ def test_describe_instance_attribute():
 
 
 @mock_ec2
-def test_warn_on_invalid_ami():
+@mock.patch(
+    "moto.ec2.models.instances.settings.ENABLE_AMI_VALIDATION",
+    new_callable=mock.PropertyMock(return_value=True),
+)
+def test_warn_on_invalid_ami(m_flag):
     if settings.TEST_SERVER_MODE:
         raise SkipTest("Can't capture warnings in server mode.")
     ec2 = boto3.resource("ec2", "us-east-1")
-    with pytest.warns(
-        PendingDeprecationWarning,
```

Comment:
Let's keep these tests. If people do not have a valid AMI, but do not have the validation enabled, they should still receive a warning